File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> umap;
        vector<vector<string>> ans;
        for (auto word : strs) {
            string sorted_word = word;
            sort(sorted_word.begin(), sorted_word.end());
            umap[sorted_word].push_back(word);
        }
        for (auto x : umap) {
            ans.push_back(x.second);
        }
        return ans;
    }
};

File: 49-group-anagrams/group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function (strs) {
    const hash = {};
    const ans = []
    strs.forEach(str => {
        const sorted = str.split('').sort().join('');
        if (!hash[sorted]) {
            hash[sorted] = [];
        }
        hash[sorted].push(str);
    })

    for (const [key, value] of Object.entries(hash)) {
        ans.push(value);
    }
    return ans;
};